src/Creator.tsx

```diff
@@ -0,0 +1,234 @@
+import YAML from 'yaml';
+import {
+  Button,
+  Flex,
+  FlexItem,
+  Grid,
+  GridItem,
+  PageGroup,
+  PageSection,
+  Title,
+} from '@patternfly/react-core';
+import './Creator.scss';
+import './components/CatalogSection.scss';
+import {
+  AllQuickStartStates,
+  QuickStart,
+  QuickStartContext,
+  QuickStartDrawer,
+  QuickStartStatus,
+  useValuesForQuickStartContext,
+} from '@patternfly/quickstarts';
+import WrappedQuickStartTile from './components/WrappedQuickStartTile';
+import CreatorWizard, {
+  CreatorWizardState,
+  EMPTY_TASK,
+} from './components/creator/CreatorWizard';
+import { itemKindMeta } from './components/creator/meta';
+import { AppContext } from './AppContext';
+
+function makeDemoQuickStart(state: CreatorWizardState): QuickStart {
+  const selectedTypeMeta =
+    state.type !== null ? itemKindMeta[state.type] : null;
+
+  return {
+    metadata: {
+      name: 'test-quickstart',
+      kind: 'QuickStarts',
+      ...(selectedTypeMeta?.extraMetadata ?? {}),
+    },
+    spec: {
+      displayName: state.title,
+      icon: null,
+      description: state.description,
+      introduction:
+        selectedTypeMeta?.hasTasks === true ? state.introduction : undefined,
+      prerequisites:
+        selectedTypeMeta?.hasTasks === true ? state.prerequisites : undefined,
+      link:
+        selectedTypeMeta?.fields?.url === true
+          ? {
+              href: state.url,
+              text: 'View documentation',
+            }
+          : undefined,
+      type:
+        selectedTypeMeta !== null
+          ? {
+              text: selectedTypeMeta.displayName,
+              color: selectedTypeMeta.tagColor,
+            }
+          : undefined,
+      durationMinutes:
+        selectedTypeMeta?.fields?.duration === true && state.duration > 0
+          ? state.duration
+          : undefined,
+      tasks:
+        selectedTypeMeta?.hasTasks === true
+          ? state.tasks.map((task) => {
+              try {
+                return {
+                  ...YAML.parse(task.yamlContent),
+                  title: task.title,
+                };
+              } catch (e) {
+                return {
+                  title: task.title,
+                  description: 'An error occurred while parsing the task.',
+                };
+              }
+            })
+          : undefined,
+    },
+  };
+}
+
+const Creator = () => {
+  const { onNavigate } = useContext(AppContext);
+
+  const [state, setState] = useState<CreatorWizardState>({
+    type: null,
+    title: '',
+    description: '',
+    bundles: [],
+    url: '',
+    duration: 0,
+    tasks: [EMPTY_TASK],
+    introduction: '',
+    prerequisites: [],
+  });
+
+  const selectedType = useMemo(() => {
+    if (state.type === null) {
+      return null;
+    }
+
+    return { id: state.type, meta: itemKindMeta[state.type] };
+  }, [state.type]);
+
+  const quickStart = useMemo<QuickStart>(
```

Comment:
`state` is defined using `useState`, so shouldn't it only be changed when `setState` is called? And this is in `useMemo` for the same reason as https://github.com/RedHatInsights/learning-resources/pull/54#discussion_r1657062701

Reply:
The state is object and its recreated on each state change, Which in this case is any action you do because the state variables are not separated. Again, you can check by adding some login or counters to the calls.